core/templates/dev/head/app.js

```diff
@@ -364,11 +364,22 @@ oppia.factory('urlService', ['$window', function($window) {
 
 // Service for computing the window dimensions.
 oppia.factory('windowDimensionsService', ['$window', function($window) {
+  var onResizeHooks = [];
+  $window.onresize = function() {
+    onResizeHooks.forEach(function(hookFn) {
+      hookFn();
+    });
+  };
   return {
     getWidth: function() {
       return (
         $window.innerWidth || document.documentElement.clientWidth ||
         document.body.clientWidth);
+    },
+    addOnResizeHook: function(hookFn) {
```

Comment:
Actually, looking at this, I think registerOnResizeHook is a bit better (since it's usual for a component to set itself up and then register stuff). Would it be OK to change this? Sorry for not realizing earlier.